### PR TITLE
feat(expo-go): expo go support

### PR DESCRIPTION
**Pull request by zetavg**
Support Expo Go by handling the [Expo Manifest Request](https://github.com/expo/expo/blob/sdk-50/docs/pages/archive/technical-specs/expo-updates-0.mdx#manifest-request) correctly.

## Background

While using Expo Go, it will first fire a [Expo Manifest Request](https://github.com/expo/expo/blob/sdk-50/docs/pages/archive/technical-specs/expo-updates-0.mdx#manifest-request) to the provided dev server, then get the bundle URL from the manifest response, and use that URL to load the app bundle.

## Implementation

* Adds a vite plugin that prepends a Connect middleware to handle the Expo manifest request.
  * Handle the Expo manifest request by dynamically importing internals from the Expo SDK installed in the user's project.
* Patch h3 to provide the original host to the proxied server - necessary for constructing the Expo manifest request, or it'll only get 127.0.0.1, and Expo Go on physical devices cannot work with that.

## Notes

* Should start the server with `--host 0.0.0.0` if connecting from Expo Go from another physical device.

## Known Issues

* Expo Go on physical devices will not work if user installs vxrn from NPM since the patch to h3 will not be included.
  * We'll eventually remove h3, so it may not be a big deal.
* It may not work with HTTPS; the reason is the same as above.

### CODE REVIEW

File: packages/vxrn/src/exports/dev.ts [hunk 1]
```diff
@@ -139,21 +137,6 @@ export const dev = async (optionsIn: VXRNOptions & { clean?: boolean }) => {
 
   app.use(router)
 
-  // TODO move these to router.get():
```
Comment (natew) [inline packages/vxrn/src/exports/dev.ts:142]: does this still work with the other examples? Seems like we may want to keep this logic for non-expo projects

Comment (zetavg) [reply packages/vxrn/src/exports/dev.ts:142]: Hmm, I'll have to try it as I haven't considered this case, thinking that Expo Go can only open expo projects anyway.

Maybe I can let the expo manifest plugin fallback to self-implemented manifest generation if it failed to load expo from user's project - haven't got it working yet, I thought I provided the necessary fields correctly but Expo Go iOS app just crash on load.

**Comment by zetavg** [comment]
More known issues:

* HMR seems not to be working on physical devices w/ Expo Go.
* Percentage of build (during the initial load) is not reported to Expo Go.

File: packages/vxrn/src/plugins/expoManifestRequestHandlerPlugin.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        // Try to dynamically import the internal Expo manifest handler from expo packages installed in the user's project.
+        let ExpoGoManifestHandlerMiddleware
+        let expoGoManifestHandlerMiddlewareImportError
+        try {
+          // Import Expo from the user's project instead of from where vxrn is installed, since vxrn may be installed globally or at the root workspace.
+          const require = module.createRequire(projectRoot)
+          const importPath = require.resolve(
+            '@expo/cli/build/src/start/server/middleware/ExpoGoManifestHandlerMiddleware.js',
+            { paths: [projectRoot] }
+          )
```
Comment (zetavg) [inline packages/vxrn/src/plugins/expoManifestRequestHandlerPlugin.ts:41]: Saw that [`import-meta-resolve`](https://www.npmjs.com/package/import-meta-resolve) is being used elsewhere, maybe we can try to use it here too.

**Pull request merged**